lib/rouge/lexers/mason.rb

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Mason < TemplateLexer
+      title 'Mason'
+      desc 'HTML::Mason'
+      tag 'mason'
+      filenames '*.mi', '*.mc', '*.mas', '*.m', '*.mhtml', '*.mcomp', 'autohandler', 'dhandler'
+      mimetypes 'text/x-mason', 'application/x-mason'
+
+      def initialize(*)
+        super
+        @perl = Perl.new
+      end
+  
+      def self.detect?(text)
+        return false if text.doctype?(/[ht|x]ml/)

```

Comment:
This regex is incorrect. `/[ht|x]ml/` will match `"|ml"` and `"thml"`  as well.
What you need is perhaps `/((?:ht|x)ml)/`

Reply:
LOL you're right. I somehow thought that `[]` was to write the choice thingy